.github/workflows/publish.yml

```
@@ -0,0 +1,30 @@
+name: Publish Crate
+
+on:
+  push:
+    tags:
+      - '*'  # Trigger when a tag is created
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  publish:
+    name: Publish to Crates.io
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        toolchain:
+          - stable
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install Rust toolchain
+        run: rustup update ${{ matrix.toolchain }} && rustup default ${{ matrix.toolchain }}
+      - name: Build Release
+        run: cargo build --release --verbose
+      - name: Test
+        run: cargo test --verbose
```

minor: let's check that documentation builds as well
```suggestion
      - name: Test
        run: cargo test --verbose
      - name: Docs
        run: cargo doc --verbose
```